packages/react-components/src/components/MentionPopover.tsx

```diff
@@ -0,0 +1,322 @@
+  useEffect(() => {
+    const rect = target?.current?.getBoundingClientRect();
+    const maxWidth = 200;
+    const finalPosition: Position = { maxWidth };
+
+    // Figure out whether it will fit horizontally
+    const leftOffset = targetPositionOffset?.left ?? 0;
+    if (leftOffset + maxWidth > (rect?.width ?? 0)) {
+      finalPosition.right = (rect?.width ?? 0) - leftOffset;
+    } else {
+      finalPosition.left = leftOffset;
+    }
+
+    if (location === 'below') {
+      finalPosition.top = (rect?.height ?? 0) + (targetPositionOffset?.top ?? 0);
+    } else {
+      // (location === 'above')
+      finalPosition.bottom = (rect?.height ?? 0) - (targetPositionOffset?.top ?? 0);
+    }
+    setPosition(finalPosition);
+  }, [location, target, targetPositionOffset]);
+
+  const handleOnKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      switch (e.key) {
+        case 'Escape':
+          onDismiss && onDismiss();
+          break;
+        default:
+          break;
+      }
+    },
+    [onDismiss]
+  );
+
+  const personaRenderer = (displayName?: string): JSX.Element => {
+    const displayNamePlaceholder = localeStrings.participantItem.displayNamePlaceholder;
+    const avatarOptions = {
+      text: displayName?.trim() || displayNamePlaceholder,
+      size: PersonaSize.size24,
+      initialsColor: theme.palette.neutralLight,
+      initialsTextColor: theme.palette.neutralSecondary,
+      showOverflowTooltip: false,
+      showUnknownPersonaCoin: !displayName?.trim() || displayName === displayNamePlaceholder
+    };
+
+    return <Persona {...avatarOptions} />;
+  };
+
+  const defaultOnRenderSuggestionItem = (
+    suggestion: Mention,
+    onSuggestionSelected: (suggestion: Mention) => void,
+    active: boolean
+  ): JSX.Element => {
+    return (
+      <div
+        data-is-focusable={true}
+        /* @conditional-compile-remove(mention) */
+        data-ui-id={ids.mentionSuggestionItem}
+        key={suggestion.id}
+        onClick={() => onSuggestionSelected(suggestion)}
+        onMouseEnter={() => setHoveredSuggestion(suggestion)}
+        onMouseLeave={() => setHoveredSuggestion(undefined)}
+        onKeyDown={(e) => {
+          handleOnKeyDown(e);
+        }}
+        className={suggestionItemWrapperStyle(theme)}
+      >
+        <Stack
+          horizontal
+          className={suggestionItemStackStyle(
+            theme,
+            hoveredSuggestion?.id === suggestion.id,
+            (changedSelection ?? false) && active
+          )}
+        >
+          {personaRenderer(suggestion.displayText)}
+        </Stack>
+      </div>
+    );
+  };
+
+  const getHeaderTitle = (): string => {
+    if (title) {
+      return title;
+    }
+    /* @conditional-compile-remove(mention) */
+    return localeStrings.mentionPopover.mentionPopoverHeader;
+    return '';
+  };
+
+  return (
+    <div ref={popoverRef}>
+      <Stack
+        className={mergeStyles(
+          {
```

Comment:
Let's move these styles into the style file

Reply:
The split here for the styles is deliberate; we need to enforce the position and location but we have exposed the other styles through the style file.

Reply:
Added parameters and an interface to the style.